src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java

```diff
@@ -87,6 +87,7 @@
     private final Settings settings;
     private final AuditConfig.Filter auditConfigFilter;
     private final String opendistrosecurityIndex;
+    protected ComplianceConfig complianceConfig;
```

Comment:
`private volatile` or `private final AtomicReference`

Reply:
I thought of doing the volatile change when subscribe is added. 
But I ll add it here itself

Reply:
If you remove setter (`setComplienceConfig()`), it can wait till subscribe is added.

Reply:
Alright. I will remove setter `setComplienceConfig`  and re-add when needed